### PR TITLE
[Dockerfile] Skip COPY of files when BINDDIR is set

### LINKED ISSUE #36413: Dockerfile: skip COPY if BIND_DIR is used
https://github.com/moby/moby/pull/35100 rewrote the dockerfile to use multi-stage  builds.

https://github.com/moby/moby/pull/35100#discussion_r170151654 made a suggestion to split the last `COPY` step to a separate stage, so that no copy is performed when using `make BIND_DIR=. ....`.

This is part of the improvement that was worked on in https://github.com/moby/moby/pull/29298

### PR DESCRIPTION
**Pull request by dnephin**
Fixes #36413

Also restore sourcing of `./.bashrc` that was lost with the change to mutli-stage.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -156,7 +156,14 @@ rpm: build ## build the rpm packages
...
 
-shell: build ## start a shell inside the build env
+shell: ## start a shell inside the build env
+	docker build \
+		${BUILD_APT_MIRROR} \
+		${DOCKER_BUILD_ARGS} \
+		-t "$(DOCKER_IMAGE)" \
+		--target dev-interactive \
+		-f "$(DOCKERFILE)" .
+	# TODO: always set BINDDIR=.
```
Comment (dnephin) [inline Makefile]: I'm not sure how to make this work. The current logic for BINDDIR is a bit messy. 

Do we really need separate bind mount for `.git` and `bundles` ?

Comment (dnephin) [reply Makefile]: Since the Makefile is not used by CI, can we default to `-v $PWD:/go/src/github.com/docker/docker` ?

Comment (thaJeztah) [reply Makefile]: to skip uploading the build context, we could pipe the Dockerfile;

```
cat Dockerfile | docker build -
```

Comment (thaJeztah) [reply Makefile]: and yes, the bind_dir is messy, think I linked to an old PR of mine from the multi-stage build PR (typing from my phone right now :sweat_smile:)

Comment (dnephin) [reply Makefile]: We could also just set it to some empty directory, but we still need the context for `hack/dockerfile` so I don't think we can do either.

Comment (thaJeztah) [reply Makefile]: Ah, yes, I recall now, that was the thing I ran into at the time (and had a dirty workaround to dynamically change the `.dockerignore`)

Comment (thaJeztah) [reply Makefile]: We need cookies to bribe Tonis in accepting https://github.com/moby/moby/issues/12886

**Comment by dnephin** [comment]
Using a Makefile to drive the docker build/run makes this unnecessarily difficult. As I've [argued in the past](https://github.com/docker/cli/pull/343) a Makefile is poorly suited to this task. I'm not going to bother fighting with it to make this work correctly.

**Pull request closed without merging**